cmd/cstor-pool-mgmt/controller/replica-controller/handler.go

```diff
@@ -637,47 +633,32 @@ func (c *CStorVolumeReplicaController) populateVersion(cvr *apis.CStorVolumeRepl
 	v := cvr.Labels[string(apis.OpenEBSVersionKey)]
 	// 1.3.0 onwards new CVR will have the field populated during creation
 	if v < v130 && cvr.VersionDetails.Status.Current == "" {
-		cvr.VersionDetails.Status.Current = v
-		cvr.VersionDetails.Desired = v
-		obj, err := c.clientset.OpenebsV1alpha1().CStorVolumeReplicas(cvr.Namespace).
-			Update(cvr)
+		cvrObj := cvr.DeepCopy()
+		cvrObj.VersionDetails.Status.Current = v
+		cvrObj.VersionDetails.Desired = v
+		cvrObj, err := c.clientset.OpenebsV1alpha1().CStorVolumeReplicas(cvrObj.Namespace).
+			Update(cvrObj)
 
 		if err != nil {
-			return nil, err
+			return cvr, err
 		}
-		klog.Infof("Version %s added on cvr %s", v, cvr.Name)
-		return obj, nil
+		klog.Infof("Version %s added on cvr %s", v, cvrObj.Name)
+		return cvrObj, nil
 	}
 	return cvr, nil
 }
 
-func isCurrentVersionValid(cvr *apis.CStorVolumeReplica) bool {
-	validVersions := []string{"1.0.0", "1.1.0", "1.2.0"}
-	version := strings.Split(cvr.VersionDetails.Status.Current, "-")[0]
-	return util.ContainsString(validVersions, version)
-}
-
-func isDesiredVersionValid(cvr *apis.CStorVolumeReplica) bool {
-	validVersions := []string{"1.3.0"}
-	version := strings.Split(cvr.VersionDetails.Desired, "-")[0]
-	return util.ContainsString(validVersions, version)
-}
-
-func upgradePath(cvr *apis.CStorVolumeReplica) string {
-	return strings.Split(cvr.VersionDetails.Status.Current, "-")[0] + "-" +
-		strings.Split(cvr.VersionDetails.Desired, "-")[0]
-}
-
 func setReplicaID(u *upgradeParams) (*apis.CStorVolumeReplica, error) {
```

Comment:
Comments to this function might be helpful

Reply:
Done